h/services/delete_user.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+from h.events import AnnotationEvent
+from h.models import Annotation, Group
+from h import storage
+
+
+class UserDeleteError(Exception):
+    pass
+
+
+class DeleteUserService(object):
+    def __init__(self, request):
+        self.request = request
+
+    def delete(self, user):
+        """
+        Deletes a user with all their group memberships and annotations.
+
+        Raises UserDeleteError when deletion fails with the appropriate error
+        message.
+        """
+
+        if Group.created_by(self.request.db, user).count() > 0:
```

Comment:
As a follow-up to this, I think it would be possible to replace this with a call to `groupids_created_by` from the `GroupService`, which would remove the need for the `created_by` class method altogether.